exercises/05.actions/05.solution.history-revalidation/server/app.js

```diff
@@ -1,66 +1,72 @@
-import bodyParser from 'body-parser'
-import busboy from 'busboy'
+import { readFile } from 'fs/promises'
+import { serve } from '@hono/node-server'
+import { serveStatic } from '@hono/node-server/serve-static'
+import { RESPONSE_ALREADY_SENT } from '@hono/node-server/utils/response'
 import closeWithGrace from 'close-with-grace'
-import compress from 'compression'
-import express from 'express'
+import { Hono } from 'hono'
+import { trimTrailingSlash } from 'hono/trailing-slash'
 import { createElement as h } from 'react'
 import {
 	renderToPipeableStream,
-	decodeReplyFromBusboy,
+	decodeReply,
 } from 'react-server-dom-esm/server'
 import { App } from '../src/app.js'
 import { shipDataStorage } from './async-storage.js'
 
 const PORT = process.env.PORT || 3000
 
-const app = express()
-app.use(compress())
-// this is here so the workshop app knows when the server has started
-app.head('/', (req, res) => res.status(200).end())
+const app = new Hono({ strict: true })
 
-app.use(express.static('public', { index: false }))
-app.use('/js/src', express.static('src'))
+app.use(trimTrailingSlash())
 
-// This just cleans up the URL if the search ever gets cleared... Not important
-// for RSCs... Just ... I just can't help myself. I like URLs clean.
-app.use((req, res, next) => {
-	if (req.query.search === '') {
-		const searchParams = new URLSearchParams(req.search)
-		searchParams.delete('search')
-		const location = [req.path, searchParams.toString()]
-			.filter(Boolean)
-			.join('?')
-		return res.redirect(302, location)
-	} else {
-		next()
-	}
-})
+app.use(
+	'/js/src/*',
+	serveStatic({
+		root: './src',
+		onNotFound(path, c) {
+			c.text('Not found', 404)
+		},
+		rewriteRequestPath: path => path.replace('/js/src', ''),
+	}),
+)
+
+app.use(
+	serveStatic({
+		root: './public',
+	}),
+	async (c, next) => {
+		await next()
+	},
```

Comment:
One weird behaviour with Hono's `serverStatic` in node is that if you don't do this, it will set the response to be a 404 and stop running other routes because it didn't find a matching file. This tells hono to keep running the other middlewares and routes even though it couldn't find a matching file in the public directory. 

I was able to clean it up a little bit by combining all the middlewares into one use call.

Reply:
UPDATE: After looking at the implementation, it turns out I can get around this by just re-ordering the middlewares so that there is one to run after this one. So that is what I've done.